bin/strata-client/src/extractor.rs

```diff
@@ -517,6 +517,17 @@ mod tests {
         (tx.into(), deposit_request)
     }
 
+    /// Generate a valid [`Buf32`] for [`XOnlyPublicKey`].
+    fn generate_valid_xonly_pk_buf32() -> Buf32 {
+        let mut arb = ArbitraryGenerator::new();
+        loop {
```

Comment:
At first this loop kind of startled me, but then I realized that the probability is ~0.5. :D

Reply:
Exactly. The probability that a random 32-byte buffer is a valid X-only public key is the probability that x³ +7 is a quadratic residue modulo p. Since each x has about a 50% chance, the probability of it being invalid is approximately 50%.